tox.ini

```diff
@@ -11,10 +11,23 @@ deps =
   lint: pylint~=2.3.1
   mypy: mypy~=0.711
 
+setenv =
+  PYTHONPATH={toxinidir}/opentelemetry-api/src/
+  mypy: MYPYPATH={env:PYTHONPATH}
+
+changedir =
+  test: opentelemetry-api/tests
+
 
 commands =
-  py37-lint: pylint opentelemetry-api/src/opentelemetry/
+; Prefer putting everything in one pylint command to profit from duplication
+; warnings.
+  py37-lint: pylint opentelemetry-api/src/opentelemetry/ opentelemetry-api/tests/
```

Comment:
I think we should have the unit tests run before the lint/mypy. When developing, I usually fix the lint issues at the end once all the code has been completed and finalized. If the build were to fail on the tests with perfect lint, after the tests are fixed, the lint might be broken due to the changes.

Reply:
I think it's better in general to fail fast. If our main use case here is CI I'd like to see errors as soon as possible.

In any case the ordering doesn't matter if we run the tests in parallel (with `-p auto`), and you can always run the unit tests alone with `-e py37-test`.

Reply:
True, the order would be defined with the `envlist` at the top.

Reply:
I also think unit tests should run before the lint/mypy step ;)

Reply:
Me too, personally I think the ordering in OpenCensus makes sense.
1. unit test
2. integration test
3. code coverage
4. lint
5. document, packaging

Reply:
I sometimes got better error messages from pylint than from my failed unit test about e.g. some non-existent variable. On the other hand I agree that spending time on making the line-breaks nice before the unit test runs through is futile. So personally I think my ordering would be:

1. mypy (what this finds are not style issues but mostly real bugs, in type annotations or code)
2. maybe pylint
3. tests (would it make sense to just run the tests *including* coverage here, or is that too slow?)
4. pylint if not run in step 2
5. flake8 (this one does not find any bugs, only style issues)
6. coverage (when making changes due to previous steps code coverage is likely to change)

In any case, like @c24t said, the ordering does not matter that much, so I'm really fine with anything.